<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>A unified interface to a plethora of gene set enrichment analysis methods • multiGSEA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="A unified interface to a plethora of gene set enrichment analysis methods">
<meta property="og:description" content="Provides a unified interface to orchestrate a plethora of GSEA
    methods over a contrast of interest as well as single sample gene set
    scoring methods. This package also provides robust interactive tools to
    explore GSEA results.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">multiGSEA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.13.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/multiGSEA.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="multigsea" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#multigsea" class="anchor"></a>multiGSEA</h1></div>
<!-- badges: start -->

<p>The <code>multiGSEA</code> package was built to facilitate the use of gene sets in the analysis of high throughput genomics data (primarily RNA-seq). Analysts can orchestrate any number of GSEA methods across a specific contrast using the unified interface provided by the <code>multiGSEA</code> function, and a shiny application is provided that facilitates the exploration and interpretation of GSEA results.</p>
<ul>
<li>The <code>multiGSEA</code> function is a wrapper that orchestrates the execution of any number of user-specified gene set enrichment analyses (GSEA) over a particular experimental contrast of interest. This will create a <code>MultiGSEAResult</code> object which stores the results of each GSEA method internally, allowing for easy query and retrieval.</li>
<li>A sister <a href="https://github.com/lianos/multiGSEA.shiny"><code>multiGSEA.shiny</code></a> package provides an <code>explore</code> function, which is invoked on <code>MultiGSEAREsult</code> objects returned from a call to <code>multiGSEA</code>. The shiny application facilitates interactive exploration of these GSEA results. This application can also be deployed to a shiny server and can be initialized by uploading a serialized <code>MultiGSEAResult</code> <code>*.rds</code> file.</li>
</ul>
<p>Full details that outline the use of this software package is provided in the <a href="https://lianos.github.io/multiGSEA/articles/multiGSEA.html">package’s vignette</a>, however a brief description is outlined below.</p>
</div>
<div id="usage" class="section level1">
<h1 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h1>
<p>A subset of the RNA-seq data tumor/normal samples in the BRCA indication from the TCGA are provided in this package. We will use that data to perform a “camera” and “fry” gene set enrichment analysis of tumor vs normal samples using the MSigDB hallmark and c2 gene set collections with <code>multiGSEA</code>.</p>
<div class="sourceCode" id="cb1"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">multiGSEA</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)
<span class="no">gdb</span> <span class="kw">&lt;-</span> <span class="fu"><a href="reference/getMSigGeneSetDb.html">getMSigGeneSetDb</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'H'</span>, <span class="st">'C2'</span>), <span class="kw">species</span> <span class="kw">=</span> <span class="st">'human'</span>, <span class="kw">id.type</span> <span class="kw">=</span> <span class="st">"entrez"</span>)
<span class="no">vm</span> <span class="kw">&lt;-</span> <span class="fu"><a href="reference/examples.html">exampleExpressionSet</a></span>(<span class="kw">dataset</span> <span class="kw">=</span> <span class="st">'tumor-vs-normal'</span>, <span class="kw">do.voom</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="no">mg</span> <span class="kw">&lt;-</span> <span class="fu"><a href="reference/multiGSEA.html">multiGSEA</a></span>(<span class="no">gdb</span>, <span class="no">vm</span>, <span class="no">vm</span>$<span class="no">design</span>, <span class="st">"tumor"</span>, <span class="kw">methods</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"camera"</span>, <span class="st">"fry"</span>))</pre></div>
<p>We can view the top “camera” results with the smallest pvalues like so:</p>
<div class="sourceCode" id="cb2"><pre class="r"><span class="fu"><a href="reference/results.html">results</a></span>(<span class="no">mg</span>, <span class="st">"camera"</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span>(<span class="no">pval</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="no">collection</span>, <span class="no">name</span>, <span class="no">padj</span>) <span class="kw">%&gt;%</span>
  <span class="no">head</span>
<span class="co">#&gt;   collection                                        name         padj</span>
<span class="co">#&gt; 1         C2      SOTIRIOU_BREAST_CANCER_GRADE_1_VS_3_UP 1.037263e-36</span>
<span class="co">#&gt; 2         C2 ROSTY_CERVICAL_CANCER_PROLIFERATION_CLUSTER 1.037263e-36</span>
<span class="co">#&gt; 3         C2         NAKAYAMA_SOFT_TISSUE_TUMORS_PCA2_DN 1.718472e-23</span>
<span class="co">#&gt; 4         C2              KANG_DOXORUBICIN_RESISTANCE_UP 1.203816e-22</span>
<span class="co">#&gt; 5         C2               CROONQUIST_IL6_DEPRIVATION_DN 2.486087e-22</span>
<span class="co">#&gt; 6         C2                     BENPORATH_PROLIFERATION 7.471399e-22</span></pre></div>
<p>The shift in expression of the genes within the top gene set can be visualized with the <code>iplot</code> function below. This plot produces interactive graphics, but rasterized versions are saved for use with this <code>README</code> file:</p>
<div class="sourceCode" id="cb3"><pre class="r"><span class="fu"><a href="reference/iplot.html">iplot</a></span>(<span class="no">mg</span>, <span class="st">'c2'</span>, <span class="st">'SOTIRIOU_BREAST_CANCER_GRADE_1_VS_3_UP'</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"density"</span>)</pre></div>
<p><img src="reference/figures/README_iplot_density.png"></p>
<div class="sourceCode" id="cb4"><pre class="r"><span class="fu"><a href="reference/iplot.html">iplot</a></span>(<span class="no">mg</span>, <span class="st">'c2'</span>, <span class="st">'SOTIRIOU_BREAST_CANCER_GRADE_1_VS_3_UP'</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"boxplot"</span>)</pre></div>
<p><img src="reference/figures/README_iplot_boxplot.png"></p>
<p>When these plots are rendered in your workspace or an Rmarkdown document, the user can hover of the genes (dots) to see their name and differential expression statistics.</p>
<p>For an immersive, interactive way to explore the GSEA results, use the <code>multiGSEA.shiny::explore(mg)</code> method!</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>The multiGSEA suite of packages will soon be submitted to bioconductor and installable via the recommended <code><a href="https://rdrr.io/pkg/BiocManager/man/install.html">BiocManager::install()</a></code> mechanism. In the meantime, install this package <em><a href="mailto:multiGSEA.shiny@develop" class="email">multiGSEA.shiny@develop</a></em> branch from github, which sould install both multiGSEA.shiny and multiGSEA packages.</p>
<div class="sourceCode" id="cb5"><pre class="r"><span class="co"># install.packages("BiocManager")</span>
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"lianos/multiGSEA.shiny@develop"</span>)</pre></div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Steve Lianoglou <br><small class="roles"> Author, maintainer </small>  </li>
<li>Denali Therapeutics <br><small class="roles"> Funder </small>  <br><small>(2018+)</small>
</li>
<li>Genentech <br><small class="roles"> Funder </small>  <br><small>(2014 - 2017)</small>
</li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/lianos/multiGSEA/actions"><img src="https://github.com/lianos/multiGSEA/workflows/R-CMD-check/badge.svg" alt="R build status"></a></li>
<li><a href="https://www.repostatus.org/#active"><img src="https://www.repostatus.org/badges/latest/active.svg" alt="Project Status: Active"></a></li>
<li><a href="https://www.tidyverse.org/lifecycle/#stable"><img src="https://img.shields.io/badge/lifecycle-stable-brightgreen.svg" alt="Lifecycle: stable"></a></li>
<li><a href="https://codecov.io/gh/lianos/multiGSEA"><img src="https://codecov.io/gh/lianos/multiGSEA/branch/develop/graph/badge.svg" alt="codecov"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Steve Lianoglou, Denali Therapeutics, Genentech.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
