<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Changelog • multiGSEA</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Changelog" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">multiGSEA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.13.15</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/multiGSEA.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      
    </div>

    <div id="multigsea-100-bioconductor-release" class="section level1">
<h1 class="page-header" data-toc-text="1.0.0">
<a href="#multigsea-100-bioconductor-release" class="anchor"></a>multiGSEA 1.0.0 (Bioconductor release)</h1>
<div id="new-features" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features" class="anchor"></a>New Features</h2>
<ul>
<li>
<p>Adds preliminary support for data-frame based inputs to <code><a href="../reference/multiGSEA.html">multiGSEA()</a></code> to more easily support pre-ranked and ORA-like tests. When <code>x</code> is a data.frame, use the <code>rank_by</code> parameter to specify the column name that holds the rank metric, and <code>rank_order</code> to specify how to order the ranks (<code>"descending"</code>, <code>"ascending"</code>, or <code>"ordered"</code>).</p>
<p>The implementation is still tied to the original design of package which only supported a numeric ranking vector, and hijacks the whole <code>xmeta.</code> workaround. Consider using <code>FacileAnalysis::ffsea(data.frame, gdb)</code> if you want a more coherent interface.</p>
</li>
</ul>
</div>
<div id="breaking-changes" class="section level2">
<h2 class="hasAnchor">
<a href="#breaking-changes" class="anchor"></a>Breaking Changes</h2>
<ul>
<li><p><code>featureId</code> used all-over-the-place and <code>sample</code> columns from single sample scoring are renamed to <code>feature_id</code> and <code>sample_id</code> to play better with facile.bio</p></li>
<li><p><code>enrichtest</code> renamed to <code>ora</code> (over representation analysis) to be more inline with nomenclature in the field</p></li>
</ul>
</div>
</div>
    <div id="multigsea-010" class="section level1">
<h1 class="page-header" data-toc-text="0.10">
<a href="#multigsea-010" class="anchor"></a>multiGSEA 0.10</h1>
<div id="user-visible-changes" class="section level2">
<h2 class="hasAnchor">
<a href="#user-visible-changes" class="anchor"></a>User Visible Changes</h2>
<ul>
<li>MSigGeneSetDb updated to v6.1 and data refactored out into GeneSetDb.* packages</li>
<li>
<code>geneSet,MultiGSEAResult</code> method does not remove duplicated columns present in both <code><a href="../reference/geneSet.html">geneSet(gdb)</a></code> and <code>logFC</code>, but rather appends a <code>*.gs</code> suffix to the duplicated columns from <code>geneSet</code>. This happened because I defined a geneset with a <code>logFC</code> column (from the experiment the set was derived) from, but this collided with (and superseded) the <code>logFC</code> value for the gene returned from <code><a href="../reference/logFC.html">logFC(mgresult)</a></code>, and messed up interpretation of a geneset’s move across a contrast.</li>
<li>Introduces idea of gene- and geneset-level metadata in a GeneSetDb. The former (gene-level metadata) was always possible by creating a GeneSetDb with a data.frame which has more columns than necessary, ie. instead of just having a collection,name,featureId data.frame, you could add a symbol column. This would simply be part of the <a href="mailto:GeneSetDb@db" class="email">GeneSetDb@db</a> data.table. If there are columns in your input data.frame that are all the same value per collection,name subset, then these are marked as “geneset level annotations” and tacked onto the <a href="mailto:GeneSetDb@table" class="email">GeneSetDb@table</a> data.table. I put this in to pave the way for simple support of “tags” per geneset, but there are other uses as well.</li>
</ul>
</div>
</div>
    <div id="multigsea-06x" class="section level1">
<h1 class="page-header" data-toc-text="0.6">
<a href="#multigsea-06x" class="anchor"></a>multiGSEA 0.6.x</h1>
</div>
    <div id="multigesa-04x" class="section level1">
<h1 class="page-header" data-toc-text="0.4">
<a href="#multigesa-04x" class="anchor"></a>multiGESA 0.4.x</h1>
<div id="user-visible-changes-2" class="section level2">
<h2 class="hasAnchor">
<a href="#user-visible-changes-2" class="anchor"></a>User Visible Changes</h2>
<ul>
<li><p>scoreSingleSamples now defualts to returning a melted data.frame of results (previously we returned a (list of) matrix of scores). <code>melted=FALSE</code> parameter was changed to <code>as.matrix=FALSE</code></p></li>
<li><p>Internal MSigDB collections updated to v5.1 (as of v0.4.30). The major difference in v5.1 vs v5.0 seems to be an updated c7 (immunologic) collection, which comes on the heels of this paper: Compendium of Immune Signatures Identifies Conserved and Species-Specific Biology in Response to Inflammation <a href="http://www.cell.com/immunity/abstract/S1074-7613(15)00532-4" class="uri">http://www.cell.com/immunity/abstract/S1074-7613(15)00532-4</a></p></li>
<li><p>All methods now default to returning data.frame(s) instead of data.table(s) data.table is still used internally, however I understand that some people don’t want to have them returned into their workspace. A global option (multiGSEA.df.return) is available for you to tweak this behavior, ie. set <code><a href="https://rdrr.io/r/base/options.html">options(multiGSEA.df.return='data.tabe')</a></code> to have <code><a href="../reference/geneSets.html">geneSets(gdb)</a></code> return a data.table instead of a data.frame.</p></li>
<li><p>Changed geneSetFeatureStatistics to geneSetsStats</p></li>
</ul>
</div>
<div id="new-features-2" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features-2" class="anchor"></a>New Features</h2>
<ul>
<li><p><code>use.treat</code> parameter has been added to relevant places to run the internal differential testing via the “treat” framework made available in limma and edgeR via the <code>treat</code> and <code>glmTreat</code> “pipelines,” respectively. By default, all pipelines do not use the treat framework (but perhaps they should).</p></li>
<li><p>goseq, fry, and romer testing methods have been added.</p></li>
<li>
<p>Support for DGEList expression input fully baked in. Previously, a DGEList was shot through voom to be used internally. Internal differential gene expression statistics on DGELists use edgeR’s quasi-likelihood framework. See <code><a href="https://rdrr.io/pkg/edgeR/man/glmQLFTest.html">?edgeR::glmQLFit</a></code> for more information, and particulary reference Aaron Lun’s tutorial on using this approach here:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a>It<span class="st">'s DE-licious: a recipe for differential expression analyses of RNA-seq</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="st">experiments using quasi-likelihood methods in edgeR</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="st">http://www.statsci.org/smyth/pubs/QLedgeRPreprint.pdf</span></span></code></pre></div>
</li>
<li><p>Adds support for PANTHER pathway (and GOSLIM) genesets from PANTHER.db package via the <code>getPantherGeneSetDb</code> function.</p></li>
<li><p>Added geneSetFeatureStats to return the logFC (and membership) information for a geneset after a multiGSEA run.</p></li>
<li><p>Added constructors and <code>as()</code> coercion functions to create GeneSetDb’s from GeneSetCollection(s) and data.frames (and vice versa).</p></li>
<li><p><code>scoreSingleSamples</code> is another wrapper function which calls several single sample geneset scoring algorithms, including the ones provided by the GSVA package (such as plage and ssGSEA)</p></li>
<li><p>Adds expression-utils.R for CPM/RPKM and meltx. Also has methods to facilitate “universal access” to pData, fData, and expression data from the different often-used expression containers (ExpressionSet, EList, DGEList, SummarizedExperiment)</p></li>
<li>
<p>MSigDB definitions were updated to v5.0, which adds a new “hallmark (h)” gene set. A call to <code>getMSigDBset</code> without a version parameter will return the v5.0 data.</p>
<p>Note that MSigDB provides the entrez IDs for each geneset as human genes. The mouse version of these gene sets was constructing by mapping the human entrez ID’s to their mouse orthologs via <code>igis::orthologs</code>. When this mapping returns multiple mouse IDs for a single human ID, all of the mouse IDs are kept in the mouse geneset.</p>
<p>To get the previous v4.0 genesets (provided by WEHI), you would simply specify version=‘v4.0’ in the getMSigDBSet like so:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="st">```</span><span class="dt">r</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="dt">gdb5 &lt;- getMSigDBsetc(c('c2', 'c7'), species='mouse')</span></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="dt">gdb4 &lt;- getMSigDBsetc(c('c2', 'c7'), species='mouse', version='v4.0')</span></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="st">````</span></span></code></pre></div>
</li>
</ul>
</div>
<div id="bug-fixes" class="section level2">
<h2 class="hasAnchor">
<a href="#bug-fixes" class="anchor"></a>Bug Fixes</h2>
<ul>
<li>do.hyperGeometricTest was failing when the expression object <code>x</code> was a matrix.</li>
</ul>
</div>
</div>
    <div id="multigsea-010-1" class="section level1">
<h1 class="page-header" data-toc-text="0.1.0">
<a href="#multigsea-010-1" class="anchor"></a>multiGSEA 0.1.0</h1>
<div id="new-features-3" class="section level2">
<h2 class="hasAnchor">
<a href="#new-features-3" class="anchor"></a>New Features</h2>
<ul>
<li><p><code>multiGSEA</code> returns a <code>MultiGSEAResult</code> object which holds all of the things one would need to analyze/plot/poke at the results of the function call.</p></li>
<li><p>Plotting functions added so user can plot results of arbitrary genesets.</p></li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by Steve Lianoglou, Denali Therapeutics, Genentech.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


