<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Perform goseq Enrichment tests across a GeneSetDb. — goseq • multiGSEA</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Perform goseq Enrichment tests across a GeneSetDb. — goseq" />
<meta property="og:description" content="Note that we do not import things from goseq directly, and only load
it if this function is fired. I can't figure out a way to selectively
import functions from the goseq package without it having to load its
dependencies, which take a long time -- and I don't want loading multiGSEA
to take a long time. So, the goseq package has moved to Suggests and then
is loaded within this function when necessary." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">multiGSEA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.99</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/multiGSEA.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform goseq Enrichment tests across a GeneSetDb.</h1>
    
    <div class="hidden name"><code>goseq.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Note that we do not import things from goseq directly, and only load
it if this function is fired. I can't figure out a way to selectively
import functions from the goseq package without it having to load its
dependencies, which take a long time -- and I don't want loading multiGSEA
to take a long time. So, the goseq package has moved to Suggests and then
is loaded within this function when necessary.</p>
    </div>

    <pre class="usage"><span class='fu'>goseq</span><span class='op'>(</span>
  <span class='va'>gsd</span>,
  <span class='va'>selected</span>,
  <span class='va'>universe</span>,
  <span class='va'>feature.bias</span>,
  method <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"Wallenius"</span>, <span class='st'>"Sampling"</span>, <span class='st'>"Hypergeometric"</span><span class='op'>)</span>,
  repcnt <span class='op'>=</span> <span class='fl'>2000</span>,
  use_genes_without_cat <span class='op'>=</span> <span class='cn'>TRUE</span>,
  plot.fit <span class='op'>=</span> <span class='cn'>FALSE</span>,
  do.conform <span class='op'>=</span> <span class='cn'>TRUE</span>,
  as.dt <span class='op'>=</span> <span class='cn'>FALSE</span>,
  .pipelined <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>gsd</th>
      <td><p>The <code>GeneSetDb</code> object to run tests against</p></td>
    </tr>
    <tr>
      <th>selected</th>
      <td><p>The ids of the selected features</p></td>
    </tr>
    <tr>
      <th>universe</th>
      <td><p>The ids of the universe</p></td>
    </tr>
    <tr>
      <th>feature.bias</th>
      <td><p>a named vector as long as <code><a href='https://rdrr.io/r/base/nrow.html'>nrow(x)</a></code> that has the
"bias" information for the features/genes tested (ie. vector of gene
lengths). <code><a href='https://rdrr.io/r/base/names.html'>names(feature.bias)</a></code> should equal <code><a href='https://rdrr.io/r/base/colnames.html'>rownames(x)</a></code>.
If this is not provided, all feature lengths are set to 1 (no bias).
The goseq package provides a <code><a href='https://rdrr.io/pkg/goseq/man/getlength.html'>getlength</a></code> function which
facilitates getting default values for these if you do not have the
correct values used in your analysis.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>The method to use to calculate the unbiased category
enrichment scores</p></td>
    </tr>
    <tr>
      <th>repcnt</th>
      <td><p>Number of random samples to be calculated when random sampling
is used. Ignored unless <code>method="Sampling"</code>.</p></td>
    </tr>
    <tr>
      <th>use_genes_without_cat</th>
      <td><p>A boolean to indicate whether genes without a
categorie should still be used. For example, a large number of gene may
have no GO term annotated. If this option is set to FALSE, those genes
will be ignored in the calculation of p-values (default behaviour). If
this option is set to TRUE, then these genes will count towards the total
number of genes outside the category being tested.</p></td>
    </tr>
    <tr>
      <th>do.conform</th>
      <td><p>By default <code>TRUE</code>: does some gymnastics to conform
the <code>gsd</code> to the <code>universe</code> vector. This should neber be set
to <code>FALSE</code>, but this parameter is here so that when this function
is called from the <code><a href='multiGSEA.html'>multiGSEA</a></code> codepath, we do not have to
reconform the <code>GeneSetDb</code> object, because it has already been done.</p></td>
    </tr>
    <tr>
      <th>.pipelined</th>
      <td><p>If this is being external to a multiGSEA pipeline, then
some additional cleanup of columns name output will be done. Otherwise
the column renaming and post processing is left to the do.goseq caller
(Default: <code>FALSE</code>).</p></td>
    </tr>
    <tr>
      <th>active.only</th>
      <td><p>If <code>TRUE</code>, only "active" genesets are used</p></td>
    </tr>
    <tr>
      <th>value</th>
      <td><p>The feature_id types to extract from <code>gsd</code></p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>data.table</code> of results, similar to goseq output. The output
from <code><a href='https://rdrr.io/pkg/goseq/man/nullp.html'>nullp</a></code> is added to the outgoing data.table as
an attribue named <code>"pwf"</code>.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Young, M. D., Wakefield, M. J., Smyth, G. K., Oshlack, A. (2010).
Gene ontology analysis for RNA-seq: accounting for selection bias.
<em>Genome Biology</em> 11, R14. http://genomebiology.com/2010/11/2/R14</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Steve Lianoglou, Denali Therapeutics, Genentech.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


