<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Performs an overrepresentation analysis, (optinally) accounting for bias. — ora • multiGSEA</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Performs an overrepresentation analysis, (optinally) accounting for bias. — ora" />
<meta property="og:description" content="This function wraps limma::kegga() to perform biased overrepresntation
analysis over gene set collection stored in a GeneSetDb (gsd) object. Its
easiest to use this function when the biases and selection criteria are
stored as columns of the input data.frame dat." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">multiGSEA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.13.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/multiGSEA.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Performs an overrepresentation analysis, (optinally) accounting for bias.</h1>
    
    <div class="hidden name"><code>ora.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function wraps <code><a href='https://rdrr.io/pkg/limma/man/goana.html'>limma::kegga()</a></code> to perform biased overrepresntation
analysis over gene set collection stored in a GeneSetDb (<code>gsd</code>) object. Its
easiest to use this function when the biases and selection criteria are
stored as columns of the input data.frame <code>dat</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>ora</span>(
  <span class='no'>gsd</span>,
  <span class='no'>dat</span>,
  <span class='kw'>selected</span> <span class='kw'>=</span> <span class='st'>"significant"</span>,
  <span class='kw'>groups</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>feature.bias</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>universe</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>restrict.universe</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>plot.bias</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='no'>...</span>,
  <span class='kw'>as.dt</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>.pipelined</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>gsd</th>
      <td><p>The GeneSetDb</p></td>
    </tr>
    <tr>
      <th>dat</th>
      <td><p>A data.frame with feature-level statistics. Minimally, this should
have a <code>"feature_id"</code> (character) column, but read on ...</p></td>
    </tr>
    <tr>
      <th>selected</th>
      <td><p>Either the name of a logical column in <code>dat</code> used to subset
out the features to run the enrichement over, or a character vector of
<code>"feature_id"</code>s that are selected from <code>dat[["feature_id"]]</code>.</p></td>
    </tr>
    <tr>
      <th>groups</th>
      <td><p>Encodes groups of features that we can use to test selected
features individual, as well as "all" together. This can be specified by:
(1) specifying a name of a column in <code>dat</code> to split the enriched features
into subgroups. (2) A named list of features to intersect with <code>selected</code>.
By default this is <code>NULL</code>, so we only run enrichment over
all elements in <code>selected</code>. See examples for details.</p></td>
    </tr>
    <tr>
      <th>feature.bias</th>
      <td><p>If <code>NULL</code> (default), no bias is used in enrichment
analysis. Otherwise, can be the name of a column in <code>dat</code> to extract
a numeric bias vector (gene length, GC content, average expression, etc.)
or a named (using featureIds) numeric vector of the same. The BiasedUrn
CRAN package is required when this is not NULL.</p></td>
    </tr>
    <tr>
      <th>universe</th>
      <td><p>Defaults to all elements in <code>dat[["feature_id"]]</code>.</p></td>
    </tr>
    <tr>
      <th>restrict.universe</th>
      <td><p>See same parameter in <code><a href='https://rdrr.io/pkg/limma/man/goana.html'>limma::kegga()</a></code></p></td>
    </tr>
    <tr>
      <th>plot.bias</th>
      <td><p>See <code>plot</code> parameter in <code><a href='https://rdrr.io/pkg/limma/man/goana.html'>limma::kegga()</a></code>. You can generate
this plot without running <code>ora</code> using the <code><a href='plot_ora_bias.html'>plot_ora_bias()</a></code>,
like so:
<code><a href='plot_ora_bias.html'>plot_ora_bias(dat, selected = selected, groups = groups, feature.bias = feature.bias)</a></code></p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A data.frame of pathway enrichment. The last N colums are enrichment
statistics per pathway, grouped by the <code>groups</code> parameter. <code>P.all</code> are the
stats for all selected features, and the remaingin <code>P.*</code> columns are for
the features specifed by <code>groups</code>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>In principle, this test does what <code>goseq</code> does, however I found that
sometimes calling goseq would throw errors within <code><a href='https://rdrr.io/pkg/goseq/man/nullp.html'>goseq::nullp()</a></code> when
calling <code>makesplines</code>. I stumbled onto this implementation when googling
for these errors and landing here:
https://support.bioconductor.org/p/65789/#65914</p>
<p>The meat and potatoes of this function's code was extracted from
<code><a href='https://rdrr.io/pkg/limma/man/goana.html'>limma::kegga()</a></code>, written by Gordon Smyth and Yifang Hu.</p>
<p>Note that the BiasedUrn CRAN package needs to be installed to support biased
enrichment testing</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Young, M. D., Wakefield, M. J., Smyth, G. K., Oshlack, A. (2010).
Gene ontology analysis for RNA-seq: accounting for selection bias.
<em>Genome Biology</em> 11, R14. http://genomebiology.com/2010/11/2/R14</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>dgestats</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='examples.html'>exampleDgeResult</a></span>(<span class='st'>"human"</span>, <span class='st'>"ensembl"</span>)
<span class='no'>gdb</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='getMSigGeneSetDb.html'>getMSigGeneSetDb</a></span>(<span class='st'>"h"</span>, <span class='st'>"human"</span>, <span class='st'>"ensembl"</span>)

<span class='co'># Run enrichmnent without accounting for any bias</span>
<span class='no'>nobias</span> <span class='kw'>&lt;-</span> <span class='fu'>ora</span>(<span class='no'>gdb</span>, <span class='no'>dgestats</span>, <span class='kw'>selected</span> <span class='kw'>=</span> <span class='st'>"selected"</span>, <span class='kw'>groups</span> <span class='kw'>=</span> <span class='st'>"direction"</span>,
              <span class='kw'>feature.bias</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)

<span class='co'># Run enrichment and account for gene length</span>
<span class='no'>lbias</span> <span class='kw'>&lt;-</span> <span class='fu'>ora</span>(<span class='no'>gdb</span>, <span class='no'>dgestats</span>, <span class='kw'>selected</span> <span class='kw'>=</span> <span class='st'>"selected"</span>,
             <span class='kw'>feature.bias</span> <span class='kw'>=</span> <span class='st'>"effective_length"</span>)

<span class='co'># plot length bias with DGE status</span>
<span class='fu'><a href='plot_ora_bias.html'>plot_ora_bias</a></span>(<span class='no'>dgestats</span>, <span class='st'>"selected"</span>, <span class='st'>"effective_length"</span>)</div><div class='img'><img src='ora-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'># induce length bias and see what is the what ...............................</span>
<span class='no'>biased</span> <span class='kw'>&lt;-</span> <span class='no'>dgestats</span>[<span class='fu'><a href='https://rdrr.io/r/base/order.html'>order</a></span>(<span class='no'>dgestats</span>$<span class='no'>pval</span>),]
<span class='no'>biased</span>$<span class='no'>effective_length</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/sort.html'>sort</a></span>(<span class='no'>biased</span>$<span class='no'>effective_length</span>, <span class='kw'>decreasing</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)
<span class='fu'><a href='plot_ora_bias.html'>plot_ora_bias</a></span>(<span class='no'>biased</span>, <span class='st'>"selected"</span>, <span class='st'>"effective_length"</span>)</div><div class='img'><img src='ora-2.png' alt='' width='700' height='433' /></div><div class='input'><span class='no'>etest</span> <span class='kw'>&lt;-</span> <span class='fu'>ora</span>(<span class='no'>gdb</span>, <span class='no'>biased</span>, <span class='kw'>selected</span> <span class='kw'>=</span> <span class='st'>"selected"</span>,
             <span class='kw'>groups</span> <span class='kw'>=</span> <span class='st'>"direction"</span>,
             <span class='kw'>feature.bias</span> <span class='kw'>=</span> <span class='st'>"effective_length"</span>)</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Steve Lianoglou, Denali Therapeutics, Genentech.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


